### PR TITLE
Added Vagrant configuration

### PR DESCRIPTION
**Pull request by 11galpid**
-Added .gitignore to ignore .vagrant folder created after using Vagrant.
-Added initialization scripts (both root and usermode scripts)
-Added startup (currently empty) script
-Added Vagrantfile with updated configuration

### CODE REVIEW

File: Vagrantfile
Comment (lmilbaum) [inline Vagrantfile]: The project files are synced automatically to /vagrant.  What is the purpose of syncing those files again?

File: Vagrantfile
Comment (lmilbaum) [inline Vagrantfile:16]: Is privileged necessary ?

File: Vagrantfile
Comment (lmilbaum) [inline Vagrantfile]: You might want to consider using the path attribute

File: init_scripts/init-root.sh
Comment (lmilbaum) [inline init_scripts/init-root.sh:1]: Please elaborate what this script is doing and why it is needed

File: selotape/startup.sh
Comment (lmilbaum) [inline selotape/startup.sh]: Do not add extra code/files as place holders.

**Review by lmilbaum — CHANGES_REQUESTED** [review]

File: Vagrantfile
Comment (11galpid) [reply Vagrantfile]: From my usage, the /vagrant folder wasn't syncing properly (Uses RSYNC by default), reading online a fix is to add a different folder, and it by default uses the VirtualBox syncing method, which works great

File: Vagrantfile
Comment (11galpid) [reply Vagrantfile:16]: Yes, this is meant to be run as root (system level configuration)

File: init_scripts/init-root.sh
Comment (11galpid) [reply init_scripts/init-root.sh:1]: Disables built-in repositories which seemed to slow down the "dnf update" procedure, adding a faster localized repository (israeli based).
Then the script actually updates the system, and installs system-level tools which can come in handy for the developer (as this cloud-based image is stripped, even "vim" is not included)_

File: selotape/startup.sh
Comment (11galpid) [reply selotape/startup.sh]: OK

File: Vagrantfile
Comment (lmilbaum) [reply Vagrantfile:16]: The provisioning is already executed as root as far as I know.

File: init_scripts/init-root.sh
Comment (lmilbaum) [reply init_scripts/init-root.sh:1]: There is a discussion over Slack whether to use a Fedora box or Ubuntu. You might want to follow that.

File: Vagrantfile
Comment (lmilbaum) [reply Vagrantfile]: Take a look at this option: https://developer.fedoraproject.org/tools/vagrant/vagrant-sshfs.html

File: Vagrantfile
Comment (lmilbaum) [reply Vagrantfile]: Don't forget to add a comment or/and emoji (thumb up) in case you have decided to follow the review comment.

File: init_scripts/init-root.sh
Comment (alonweissfeld) [reply init_scripts/init-root.sh:1]: [From the Vagrant docs](https://www.vagrantup.com/docs/boxes#official-boxes), the HashiCorp (makers of vagrant) and Bento boxes are the only two recommended box sets.  

I've set up the new `bento/ubuntu-20.04` vm box and the sync folder works perfectly -  no need to re-`vagrant up` to copy new data from the synced folder, and it's already comes with essential tools such as vim. I think we should use this box (or a different long time support box) to help us simplify our configuration at the moment.

Also, this box is [supported by GitHub Actions](https://docs.github.com/en/actions/reference/virtual-environments-for-github-hosted-runners) as discussed in slack.

File: Vagrantfile
Comment (11galpid) [reply Vagrantfile:16]: Yes, but because of the fact that one of the scripts needs root and the other doesn't, I feel like it should be explicitly stated.

File: init_scripts/init-root.sh
Comment (11galpid) [reply init_scripts/init-root.sh:1]: From my testing, it has a few benefits, but also a few drawbacks.
Benefits:
- /vagrant works without any need for additional configuration
- Future proof for use with Github Actions

Drawbacks:
- "Heavy" box relative to the "cloud-base" (5min+- to download compared to 10seconds)
- Still has packages missing ("python3-pip" for instance is missing)

Overall, it seems worth the move for the Github Actions, so i'll push an update that uses this, but changes are gonna be made to all of the files included to support this.

File: Vagrantfile
Comment (11galpid) [reply Vagrantfile]: Thanks, this looks good, but its similar to the configuration I've set before, just uses a different method for the file system of the storage (ssh-fs).
Anyway, I'm updating the machine to Ubuntu (as described in one of the other comments), and the default /vagrant folder works fine in there, so we can remove this shared folders in the next push.

File: init_scripts/init-root.sh
Comment (alonweissfeld) [inline init_scripts/init-root.sh]: Can you elaborate why are we writing both to standard output and the file `resolve.conf`? What is the purpose of this file? maybe it's worth adding a comment

**Review by alonweissfeld — COMMENTED** [review]
@11galpid Looks good! 👏  I think it's a good simplification change for now.

**Comment by lmilbaum** [comment]
When the PR is ready for a re-review, please pick the re-request review symbol next to the reviewers names

File: init_scripts/init-root.sh
Comment (11galpid) [reply init_scripts/init-root.sh]: 8.8.8.8 is Google's DNS server, this line basically resets the dns server to avoid issues with virtualboxe's default nat assigned dns server (which sometimes causes issues, as it did on my machine)

**Comment by lmilbaum** [comment]
@ifireball I have added you as a reviewer.  Take a look.

File: .gitignore
Comment (ifireball) [inline .gitignore]: See the "no-newline at the end of file" warning above - please fix that - there should be an appropriate setting for your text editor.

File: Vagrantfile
Comment (ifireball) [inline Vagrantfile]: Using Ubuntu will limit our ability to assist you with system issues. Please consider using Fedora or CentOS.

File: init_scripts/init-root.sh
Comment (ifireball) [inline init_scripts/init-root.sh]: This would make your deveopment environment not reproducible. You will get a different environment every time you run the Vagrant deployment. Please avoid doing that.

File: init_scripts/init-root.sh
Comment (ifireball) [reply init_scripts/init-root.sh]: You really shouldn't mess with that - its one of those things that may break on other people's machines.

File: init_scripts/init-root.sh
Comment (ifireball) [inline init_scripts/init-root.sh]: you should try to use a fixed version so that you're guaranteed to have the same environment every time you start it.

File: init_scripts/init-user.sh
Comment (ifireball) [inline init_scripts/init-user.sh]: please learn about Python virtual environments and use them to manage your Python dependencies. I recommend using one of the modern Python dependency management tools such as `Pipenv` or `Poetry`, but you can use `requirements.txt`.

**Review by ifireball — CHANGES_REQUESTED** [review]
This change includes many different and unrelated thongs in it - they should generally be sent in different PRs, but since it ends up being a very small Pr overall, we can accept that for now.

File: Vagrantfile
Comment (alonweissfeld) [reply Vagrantfile]: But as discussed in Slack, wouldn't it be better to choose a more "out of the box" box that GitHub actions support?
Correct me if I'm wrong, using Docker to run Fedora won't replace the usage of Vagrant?

Comment (ifireball) [reply Vagrantfile]: Don't confuse GitHub actions and Vagrant. There is a an up to date Fedora Vagrant box that is officially released and supported by the Fedora project. For Vagrant you should use that.

With github actions - if you use the GitHub provided VMs - they are pre-installed with some Ubuntu version simply because that's what the GH engineers chose to provision their AZURE VMs with - but they've included the container support feature precisely to make that choice meaningless. 

As Red Hatters we work with Fedora, RHEL and CentOS all the time, we will be much better equipped to help you solve issues that arise on a Fedora environment then if those issue arise on Ubuntu or some other environment.

Comment (11galpid) [reply Vagrantfile]: Ok then, Im moving back to fedora/32-cloud-base

File: init_scripts/init-user.sh
Comment (11galpid) [reply init_scripts/init-user.sh]: Read about pipenv, will change the django install to user that (and future packages will be added to it as well).
Just a note, to install pipenv, we still need to use pip install,  however this time I use a specific pipenv version to stay consistent.

**Review by ifireball — CHANGES_REQUESTED** [review]
"PR review response" is not a good commit message - will it be helpful for you when you see it in `git log` ?

I suggest you squash all the commits in this PR to a single will-documented commit.

With all the discussion threads, I've lost track of what needs work and what is ok for now,  so once you'll submit the new commit we'll review from scratch.

File: Vagrantfile
Comment (ifireball) [reply Vagrantfile:16]: @lioramilbaum can we resolve this conversation?

File: init_scripts/init-root.sh
Comment (ifireball) [reply init_scripts/init-root.sh:1]: @11galpid , @lioramilbaum I think we can resolve this conversation for now no?

File: init_scripts/init-user.sh
Comment (ifireball) [inline init_scripts/init-user.sh:1]: this is a bad comment - comments should not describe *what* you do, comments should descry *why* you do it or things that could be not obvious about it.

The header section of a script should also include its name and details about its purpose, when, how and by whom it is executed.

**Review by ifireball — CHANGES_REQUESTED** [review]
This change should probably also a README update telling other developers about the existence of Vagrant and how to use it to bring up the development environment.

**Review by ifireball — APPROVED** [review]
LGTM - pending re-review from @lioramilbaum

**Review by lmilbaum — APPROVED** [review]
LGTM

**Pull request merged**
